tests/system/Log/Handlers/FileHandlerTest.php

```diff
@@ -77,6 +78,100 @@ public function testHandleCreateFile()
+        $this->assertSame($expectedResult, $result);
+    }
+
+    public function testHandleWithInteger()
+    {
+        $config                                       = new LoggerConfig();
+        $config->handlers[TestHandler::class]['path'] = $this->start;
+        $logger                                       = new MockFileLogger($config->handlers[TestHandler::class]);
+
+        $logger->setDateFormat('Y-m-d');
+        $date     = date('Y-m-d');
+        $expected = 'log-' . $date . '.log';
+        $logger->handle('info', 123456);
+
+        $fp     = fopen($config->handlers[TestHandler::class]['path'] . $expected, 'rb');
+        $result = stream_get_contents($fp);
+        fclose($fp);
+
+        // did the log file get created?
+        $expectedResult = 'INFO - ' . $date . " --> 123456\n";
+        $this->assertSame($expectedResult, $result);
+    }
+
+    public function testHandleWithArray()
+    {
+        $config                                       = new LoggerConfig();
+        $config->handlers[TestHandler::class]['path'] = $this->start;
+        $logger                                       = new MockFileLogger($config->handlers[TestHandler::class]);
+
+        $logger->setDateFormat('Y-m-d');
+        $date    = date('Y-m-d');
+        $arrData = [
+            'firstName' => 'John',
+            'lastName'  => 'Doe',
+        ];
+        $expected = 'log-' . $date . '.log';
+        $logger->handle('info', print_r($arrData, true));
+
+        $fp     = fopen($config->handlers[TestHandler::class]['path'] . $expected, 'rb');
+        $result = stream_get_contents($fp);
+        fclose($fp);
+
+        // did the log file get created?
+        $expectedResult = 'INFO - ' . $date . " --> Array\n(
+    [firstName] => John
+    [lastName] => Doe\n)\n\n";
+        $this->assertSame($expectedResult, $result);
+    }
+
+    public function testHandleWithObject()
+    {
+        $config                                       = new LoggerConfig();
+        $config->handlers[TestHandler::class]['path'] = $this->start;
+        $logger                                       = new MockFileLogger($config->handlers[TestHandler::class]);
+
+        $logger->setDateFormat('Y-m-d');
+        $date = date('Y-m-d');
+
+        $obj            = new stdClass();
+        $obj->firstName = 'John';
+        $obj->lastName  = 'Doe';
+
+        $expected = 'log-' . $date . '.log';
+        $logger->handle('info', print_r($obj, true));
+
+        $fp     = fopen($config->handlers[TestHandler::class]['path'] . $expected, 'rb');
+        $result = stream_get_contents($fp);
+        fclose($fp);
+
+        // did the log file get created?
+        $expectedResult = 'INFO - ' . $date . " --> stdClass Object\n(
+    [firstName] => John
+    [lastName] => Doe\n)\n\n";
+        $this->assertSame($expectedResult, $result);
+    }
+
```

Comment:
This formatting of `$message` was from commit https://github.com/codeigniter4/CodeIgniter4/commit/4fcb7607b6a082cf51cdd053f95a940abbc17ced. However, I think this is in contradiction to this line:
https://github.com/codeigniter4/CodeIgniter4/blob/53f17e8ff080b1490cd7df3c57ec2924c858e21b/system/Log/Logger.php#L44

Also, this is not compliant with PSR-3 considering that the `Logger` is implementing `Psr\Log\LoggerInterface`.

All in all, I believe this behavior is a bug to begin with.

Reply:
What about now?

Reply:
I did not know that.

```php
log_message('error', $this); // TypeError
$this->logger->error($this); // Okay. but PHPStan error
```

```
 ------ --------------------------------------------------------------------------------------------------------------------- 
  Line   app/Controllers/Home.php                                                                                             
 ------ --------------------------------------------------------------------------------------------------------------------- 
  9      Parameter #2 $message of function log_message expects string, $this(App\Controllers\Home) given.                     
  10     Parameter #1 $message of method Psr\Log\LoggerInterface::error() expects string, $this(App\Controllers\Home) given.  
 ------ --------------------------------------------------------------------------------------------------------------------- 
```

> Also, this is not compliant with PSR-3 considering that the Logger is implementing Psr\Log\LoggerInterface.
> All in all, I believe this behavior is a bug to begin with.

Yes, but fixing this may break existing apps.

Reply:
@kenjis Before changes in this PR appear PHP 2 errors. With condition now PR, what about ignoring this 2 error phpstan?

Reply:
Ignoring PHPStan errors is very bad practice. 
We should remove phpstan-baseline.neon.dist entries.

Reply:
Please revert.